TROUBLESHOOTING.md

```diff
@@ -33,20 +36,103 @@ However, if you need to declare dependency on gRPC yourself, we recommend doing
   </dependencies>
 </dependencyManagement>
 ```
-so that versions cannot go out of sync.
 
-See [gRPC documentation](https://github.com/grpc/grpc-java/blob/master/SECURITY.md#troubleshooting)
-for more details.
+The `grpc-netty-shaded` dependency avoids conflicts with other `netty` dependencies that may also be in the classpath.
+
+If you are using `google-cloud-java` version 0.35.0 or above, then it already uses `grpc-netty-shaded`. If you are still running into `ALPN` related problems, please see [gRPC Troubleshooting guide](https://github.com/grpc/grpc-java/blob/master/SECURITY.md#troubleshooting) for other causes.
+
+## ClassNotFoundException, NoSuchMethodError, NoClassDefFoundError
+
+These errors are usually caused by having multiple versions or conflicting versions of the same dependency in the classpath.
+Usually these dependency conflicts occur with `guava` or `protobuf-java`.
+
+There may be multiple sources for classpath conflicts:
+- Multiple transitive dependencies in the dependency tree
```

Comment:
Better: "Multiple versions of the same transitive dependency in the dependency tree"

Reply:
good catch! fixed